blocks/developers-toc/developers-toc.js

```diff
@@ -0,0 +1,32 @@
+import ComponentBase from '../../scripts/component-base.js';
+
+const sitePathPrefix = 'developers';
+const henkelOrg = 'henkeldx';
+
+export default class DeveloperToc extends ComponentBase {
+  ready() {
+    this.setupToc();
+  }
+
+  async setupToc() {
+    const response = await fetch(`/${sitePathPrefix}/query-index.json`);
+    if(!response.ok) return;
+    const json = await response.json();
+    const repositories = json.data.reduce((map, page) => {
+      const [, , project, repository] = page.path.split('/');
+      if(!map[project]) map[project] = new Set();
+      map[project].add(repository);
+      return map;
+    }, {});
+    const generatePages = (project, repository) => {
+      const pages = json.data.filter((page) => page.path.startsWith(`/${henkelOrg}/${project}/${repository}`));
```

Comment:
Just a suggestion for future.
I remember I saw that we can add different sheets in the query-index xlsx to filter data from the query sheet with excel functionality and then we can fetch pre filtered data from that sheet only.

Would be nice and helpful if we could have a query-index also for preview to use in development... but I remember was not available.

Reply:
good point but in this case I cannot use the static filtering in excel as I need to filter dynamically by project and repository. 

I am actually not sure how query indexes on preview behave, would be interesting to investigate. The one I am using here is already there, I just cannot access it from a feature branch because of CORS because it lives on the git URL: https://main--raqn-docs-git--henkel.aem.live/query-index.json
on prod we make it available through CDN under the same domain so there it will work: https://docs.raqn.io/developers/query-index.json